crates/sozo/src/commands/options/account.rs

```diff
@@ -64,6 +63,11 @@ impl AccountOptions {
     }
 
     fn signer(&self, env_metadata: Option<&Environment>) -> Result<LocalWallet> {
+        // INVARIANT: We assume that from commandline we can either specify `private_key` along with
+        // `account_address` or `keystore_path` along with `keystore_password`.
+        //
+        // This is enforced by clap.
+        // TODO: See if there is any way to test it.
```

Comment:
You should be able to test with something like:
```rust
    #[test]
    fn test_bad_args() {
        assert!(Command::try_parse_from(["sozo", "--keystore", "/path", "--private-key", "0x1"]).is_err());                                             
    }
```

Reply:
This would fail only because of no account address, but not because of both keystore and private key are provided. Right now it would just silently ignore the keystore and the private key would take precedence if both are provided. 

What do you guys think about this? Should it return an error if both are provided? or  maintain current logic?

I could imagine some users might get confused if both are set (eg in `Scarb.toml`) and not knowing which signer is being used.

Ideally, I'd prefer if `sozo` can output the source of these parameters (either cli or `Scarb.toml`), but that is something for future discussion.

Reply:
Anyway, we should be able to test this flow following what @glihm suggested.

Reply:
> This would fail only because of no account address, but not because of both keystore and private key are provided. Right now it would just silently ignore the keystore and the private key would take precedence if both are provided.

Yes totally, you're right. Was more an example and definitely must be adjusted as needed.

> Ideally, I'd prefer if sozo can output the source of these parameters (either cli or Scarb.toml), but that is something for future discussion.

100%, this will avoid any ambiguity and user will learn about priorities on the go.

Reply:
> What do you guys think about this? Should it return an error if both are provided? or maintain current logic?

On CLI it won't allow providing both due to clap constraints but yes i agree in `Scarb.toml` user can specify both and `private_key` will take precedence.

For now adding a comment to make it clearer for reader of code.

Reply:
`TODO` maybe removed.

Reply:
done